pkg/controller/ads/ads_processor.go

```diff
@@ -157,17 +163,13 @@ func (p *processor) handleCdsResponse(resp *service_discovery_v3.DiscoveryRespon
 	}
 	if len(removed) > 0 {
 		log.Debugf("removed cluster: %v", removed.UnsortedList())
-		p.Cache.ClusterCache.Delete()
 	}
 
-	// TODO(hzxuzhonghu): consider flush all the clusters except those of type EDS
-	// So DNS typed clusters can take effect immediately
-
-	// Only flush the cache when there is no eds cluster
-	// Eds cluster should always be flushed in the eds handler
-	if len(p.Cache.edsClusterNames) == 0 {
```

Comment:
Can the judgment logic be retained to avoid introducing a new ApiStatus_WAITING state?

Reply:
I am introducing a new Status, it can be shared with DNS resolve